File: android/app/src/main/kotlin/com/abstractvision/phonedetox/launcher/HomeRoleRequestResult.kt
package com.abstractvision.phonedetox.launcher

enum class HomeRoleRequestResult(val wireValue: String) {
    GRANTED("granted"),
    DENIED("denied"),
    CANCELLED("cancelled"),
    ALREADY_HELD("alreadyHeld"),
    OPENED_SETTINGS("openedSettings"),
    UNAVAILABLE("unavailable"),
}

File: lib/features/launcher/domain/home_role_request_result.dart
enum HomeRoleRequestResult {
  granted,
  denied,
  cancelled,
  alreadyHeld,
  openedSettings,
  unavailable;

  static HomeRoleRequestResult fromNative(Object? value) => switch (value) {
    'granted' => HomeRoleRequestResult.granted,
    'denied' => HomeRoleRequestResult.denied,
    'cancelled' => HomeRoleRequestResult.cancelled,
    'alreadyHeld' => HomeRoleRequestResult.alreadyHeld,
    'openedSettings' => HomeRoleRequestResult.openedSettings,
    'unavailable' => HomeRoleRequestResult.unavailable,
    _ => throw FormatException(
      'Unknown native Home-role request result: $value',
    ),
  };
}
